src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -74,6 +75,13 @@ public void onEnter() {
 
         this.lastCommitMillis = System.currentTimeMillis();
         scheduleTask(this::checkCommitTimeout, getParameters().commitTimeoutMillis, TimeUnit.MILLISECONDS);
+
+        getLog().debug("Register cursor reset listener");
+        cursorResetSubscription = getZk().subscribeForCursorsReset(() -> {
+            final String message = "Resetting subscription cursors";
+            sendMetadata(message);
+            shutdownGracefully(message);
+        });
```

Comment:
There is a task queue, which guaranties, that all actions won't modify and data, and it is better to respect this behavior everywhere. 
So it is better to use `addTask` call while getting subscription reset notification.